### ISSUE TITLE
Replace BoltDB queries with Storm

### ISSUE THREAD

**Issue by devtomnl** [dusk-network/dusk-blockchain#688]
https://github.com/asdine/storm

```
Storm is a simple and powerful toolkit for BoltDB. Basically, Storm provides indexes, a wide range of methods to store and fetch data, an advanced query system, and much more.
```

**Comment by devtomnl** [comment]
motivation:
To maintain the existing code is way too much work, so using Storm things are way more simplified, specially for finding operations.

**Comment by julesdesmit** [comment]
Closed via #689
